src/java.base/share/classes/sun/security/jca/ProviderConfig.java

```diff
@@ -160,66 +160,73 @@ public String toString() {
      * Get the provider object. Loads the provider if it is not already loaded.
      */
     @SuppressWarnings("deprecation")
-    synchronized Provider getProvider() {
+    Provider getProvider() {
         // volatile variable load
         Provider p = provider;
         if (p != null) {
             return p;
         }
-        if (shouldLoad() == false) {
-            return null;
-        }
+        // DCL
+        synchronized (ProviderConfig.class) {
```

Comment:
synchronized (this) instead since 'provider' is an instance field?

Reply:
yes!